### PR TITLE
Fix memory leaks related to the glTF loader

### PR DESCRIPTION
**Pull request by bghgary**
There are probably some more to fix, but this should fix most of the cases.

### CODE REVIEW

**Comment by bghgary** [comment]
I searched a bit and it doesn't sound like there is a way to check for memory leaks. Any suggestions on how to automate testing this would be great.

File: src/Tools/babylon.promise.ts [hunk 1]
```diff
@@ -22,36 +22,6 @@ module BABYLON {
         private _onRejected?: (reason: any) => void;
         private _rejectWasConsumed = false;
 
-        public get state(): PromiseStates {
```
Comment (deltakosh) [inline src/Tools/babylon.promise.ts]: Why removing them?

Comment (bghgary) [reply src/Tools/babylon.promise.ts]: Not sure why they are needed as public. It is not part of the promise contract right?

Comment (deltakosh) [reply src/Tools/babylon.promise.ts]: Right. I used them mostly for debugging purpose. We can remove them no problem 
I was more asking to understand if it was related to memory leak

Comment (bghgary) [reply src/Tools/babylon.promise.ts]: Yeah, I figured it was something like that. I removed them to see if they were the cause of any issues. I can put them back if you want.

Comment (deltakosh) [reply src/Tools/babylon.promise.ts]: Nope this is fine like that

**Pull request merged**